### PR TITLE
(tests): Add test for "group parameters" (breakout room pre-selection)

### LINKED ISSUE #24002: Add test for group parameter (breakout room pre-selection)
**Describe the test scenario, be specific**
Test the “predefined groups” functionality when creating breakout rooms, ensuring that users are assigned according to the groups payload sent when creating a meeting, and that the client UI displays these groups correctly. 

Steps to reproduce:
1. Start a meeting with several users (e.g., users A, B, C, D).
2. Pre-configure several participant groups in the meeting settings (ensure pre-defined groups are set up).
3. During the meeting, click on the “Breakout Rooms” button.
4. In the Breakout Rooms dialog, select the option to use pre-defined groups when assigning participants.
5. Create the breakout rooms.
6. Check that participants are assigned to their correct, pre-defined groups within the breakout rooms

**Edge cases**
- Verify behavior if a participant joins or leaves the meeting while breakout rooms are being created.

**Additional context**
- This test is to verify that the feature for assigning pre-defined groups to breakout rooms works as expected
-  [API documentation](https://docs.bigbluebutton.org/development/api/#get-post-create)  specifies how to create breakout rooms with predefined groups.

### PR DESCRIPTION
**Pull request by pinhaum**
### What does this PR do?
This PR adds tests for creating a meeting with group parameters, setting a pre-selection when creating breakout rooms.
The tests verify that the group settings are being applied while creating breakout rooms

Create Params example:
````
groups=[
  {"id":"1",name:"Room 1",roster:["1235"]},
  {"id":"2",name:"Room 2",roster:["2333","2335"]},
  {"id":"3",roster:[]}
]
````

#### Video Example

- moderator

[mod-group-params.webm](https://github.com/user-attachments/assets/d4e89853-7504-483b-8339-1e5b18940bc9)

- attendee1

[atd1-group-params.webm](https://github.com/user-attachments/assets/0051cdda-177b-48d1-8464-344a75e4d5a4)

- attendee2

[atd2-group-params.webm](https://github.com/user-attachments/assets/e2ead330-8af9-4118-8dee-8e77677e9574)

- attendee1 breakout room

[br-atd1-group-params.webm](https://github.com/user-attachments/assets/8603c650-42c6-4f67-8cbc-16ed055066b7)

- attendee2 breakout room

[br-atd2-group-params.webm](https://github.com/user-attachments/assets/6dba8173-d5ce-4f59-be67-2344c42e6df9)

### Closes Issue(s)

Closes #24002 

### How to test

1. Start a BigBLueButton meeting setting custom parameters as the example above 
2. As the moderator, start breakout rooms
3. Verify that the attendees are already sorted to a Room with the pre defined 'room name'
4. start the breakout rooms
5. each attendee will be set in the pre defined room
6. run full suite: `npm run test parameter` or `npx playwright test parameters`
- you can also run specific test with:
  `npm run test:filter "Predefined groups for Breakout Rooms"`

### CODE REVIEW

File: bigbluebutton-tests/playwright/parameters/customparameters.js [hunk 1]
```diff
@@ -547,6 +547,50 @@ class CustomParameters extends MultiUsers {
...
+    await this.modPage.waitAndClick(e.createBreakoutRooms);
+
+    await this.modPage.hasElement(e.breakoutBox0, 'should display the breakout box for unassigned users,');
+
+    await this.modPage.hasElement(e.breakoutBox1, 'should display the breakout box for room 1');
+    await this.modPage.hasElementCount(`${e.breakoutBox1} > p`, 1, 'should display only 1 user in the breakout room 1');
+    await this.modPage.hasValue(e.roomNameInput, 'Room 1', 'should display the correct name of the group for breakout room 1');
+
+    await this.modPage.hasElement(e.breakoutBox2, 'should display the breakout box for room 2');
+    await this.modPage.hasElementCount(`${e.breakoutBox2} > p`, 1, 'should display only 1 user in the breakout room 2');
```
Comment (antonbsa) [inline bigbluebutton-tests/playwright/parameters/customparameters.js]: Additionally, it would be good to check if the user displayed on each breakout box is the expected one - by the username

```suggestion
    await this.modPage.hasElementCount(`${e.breakoutBox1} > p`, 1, 'should display only 1 user in the breakout room 1');
    await this.modPage.hasText(e.breakoutBox1, this.userPage.username, 'should display the correct user assigned to breakout room 1');
    await this.modPage.hasValue(e.roomNameInput, 'Room 1', 'should display the correct name of the group for breakout room 1');

    await this.modPage.hasElement(e.breakoutBox2, 'should display the breakout box for room 2');
    await this.modPage.hasElementCount(`${e.breakoutBox2} > p`, 1, 'should display only 1 user in the breakout room 2');
    await this.modPage.hasText(e.breakoutBox2, this.userPage2.username, 'should display the correct user assigned to breakout room 2');

```

File: bigbluebutton-tests/playwright/parameters/customparameters.js [hunk 2]
```diff
@@ -547,6 +547,50 @@ class CustomParameters extends MultiUsers {
...
+    await this.userPage.waitAndClick(e.modalConfirmButton);
+    await this.userPage.waitAndClick(e.breakoutRoomsItem);
+    await this.userPage.hasElement(e.alreadyConnected, 'should display the element alreadyConnected', ELEMENT_WAIT_EXTRA_LONG_TIME);
+
+    const breakoutUserPage = await this.userPage.getLastTargetPage(this.context);
+    await breakoutUserPage.closeAudioModal();
+    await breakoutUserPage.hasElementCount(e.currentUser, 1, 'should contain the current user after joining the breakout room');
+    await breakoutUserPage.hasText(e.presentationTitle, 'Room 1', 'should display the correct breakout room name');
+
+    await this.userPage2.hasElement(e.modalConfirmButton, 'should appear the modal confirm button to join breakout');
```
Comment (antonbsa) [inline bigbluebutton-tests/playwright/parameters/customparameters.js]: Additionally, it would also be good to add an assertion for the room name displayed in the full-screen confirmation modal for each user

```suggestion
    await this.userPage.hasElement(e.modalConfirmButton, 'should appear the modal confirm button to join breakout');
    await this.userPage.hasText(e.fullscreenModal, 'Room 1', 'should display the correct breakout room name in the modal');
    await this.userPage.waitAndClick(e.modalConfirmButton);
    await this.userPage.waitAndClick(e.breakoutRoomsItem);
    await this.userPage.hasElement(e.alreadyConnected, 'should display the element alreadyConnected', ELEMENT_WAIT_EXTRA_LONG_TIME);

    const breakoutUserPage = await this.userPage.getLastTargetPage(this.context);
    await breakoutUserPage.closeAudioModal();
    await breakoutUserPage.hasElementCount(e.currentUser, 1, 'should contain the current user after joining the breakout room');
    await breakoutUserPage.hasText(e.presentationTitle, 'Room 1', 'should display the correct breakout room name');

    await this.userPage2.hasElement(e.modalConfirmButton, 'should appear the modal confirm button to join breakout');
    await this.userPage2.hasText(e.fullscreenModal, 'Room 2', 'should display the correct breakout room name in the modal');

```

File: bigbluebutton-tests/playwright/parameters/customparameters.js [hunk 3]
```diff
@@ -547,6 +547,50 @@ class CustomParameters extends MultiUsers {
...
+    await this.userPage2.waitAndClick(e.breakoutRoomsItem);
+    await this.userPage2.hasElement(e.alreadyConnected, 'should display the element alreadyConnected', ELEMENT_WAIT_EXTRA_LONG_TIME);
+
+    const breakoutUserPage2 = await this.userPage2.getLastTargetPage(this.context);
+    await breakoutUserPage2.closeAudioModal();
+    await breakoutUserPage2.hasElementCount(e.currentUser, 1, 'should contain the current user after joining the breakout room');
+    await breakoutUserPage2.hasText(e.presentationTitle, 'Room 2', 'should display the correct breakout room name');
+
+    await this.modPage.waitAndClick(e.breakoutRoomsItem);
+    await this.modPage.hasText(e.userNameBreakoutRoom, /Attendee/, 'should have the attendee name on the breakout room below a room on the main breakout panel');
```
Comment (antonbsa) [inline bigbluebutton-tests/playwright/parameters/customparameters.js]: prefer to use literal strings when comparing this type of value, as it might return false positives since both users contain "Attendee" in the username

```suggestion
    await this.modPage.hasText(e.userNameBreakoutRoom, this.userPage.username, 'should have the attendee name on the breakout room below a room on the main breakout panel');

```

**Review by antonbsa — CHANGES_REQUESTED** [review]
Overall, it looks very good with concise assertions! I've just added a few improvements to some assertions:

File: bigbluebutton-tests/playwright/parameters/customparameters.js [hunk 4]
```diff
@@ -547,6 +547,57 @@ class CustomParameters extends MultiUsers {
...
+    await this.modPage.hasElement(e.breakoutBox1, 'should display the breakout box for room 1');
+    await this.modPage.hasElementCount(`${e.breakoutBox1} > p`, 1, 'should display only 1 user in the breakout room 1');
+    await this.modPage.hasText(e.breakoutBox1, this.userPage.username, 'should display the correct user assigned to breakout room 1');
+    await this.modPage.hasValue(e.roomNameInput, 'Room 1', 'should display the correct name of the group for breakout room 1');
+
+    await this.modPage.hasElement(e.breakoutBox2, 'should display the breakout box for room 2');
+    await this.modPage.hasElementCount(`${e.breakoutBox2} > p`, 1, 'should display only 1 user in the breakout room 2');
+    await this.modPage.hasText(e.breakoutBox2, this.userPage2.username, 'should display the correct user assigned to breakout room 2');
+
+    await this.modPage.hasValue(`input[data-test=roomName-2]`, 'Room 2', 'should display the correct name of the group for breakout room 2');
```
Comment (antonbsa) [inline bigbluebutton-tests/playwright/parameters/customparameters.js]: `e.roomNameInput` refers to roomName-1. I'd suggest updating this variable name to `e.roomNameInput1` and creating the `e.roomNameInput2`. Avoid using hardcoded selectors

**Review by antonbsa — CHANGES_REQUESTED** [review]

File: bigbluebutton-tests/playwright/parameters/customparameters.js (hunk 4 above)
Comment (pinhaum) [reply bigbluebutton-tests/playwright/parameters/customparameters.js]: will do!

**Review by antonbsa — APPROVED** [review]
